### PR TITLE
Ps14/remaining models

### PR DESCRIPTION
**Pull request by JorgePS14**
Created addUser, addBook, addListing endpoints. Fixed an issue that was causing Flask initialization to create a TinyBLOB column in MySQL instead of a LongBLOB for images.

How to test the endpoints:
If possible, please use Postman. You shall create registries in order for this to work. You must first delete all the tables previously created in your local instance of the MySQL database (user, book, listing) so that Flask can create new ones with the correct data types.

1. You will create one user registry by sending a POST request with a form-data body containing the following fields:
- email (string)
- password (string)
- username (string)
- location (string)
- university (string)
- semester (int)
- major (string)
Example (route is now /api/user instead of addUser):
image

2. You will create one book registry by sending a POST request with a form-data body containing the following fields:
- name (string)
- author (string)
- edition (int)
- value (float)
- isbn (string)
Example (route is now /api/book instead of addBook):
image

3. After creating one book and one user (or as many users and books as you desire, not really a problem), you are now able to create a listing registry by sending a POST request with a form-data body containing the following fields:
- photo (file)
- description (string)
- condition (int)
- no_available (int)
- price (float)
- user_id (int) // NOTE: there must be at least one user in the database with this user id, otherwise, you will get a 500 code Internal Server Error)
- book_id (int) // NOTE: there must be at least one book in the database with this book id, otherwise, you will get a 500 code Internal Server Error)
Example (route is now /api/listing instead of addListing):
image

### CODE REVIEW

File: app/routes.py [hunk 1]
```diff
@@ -11,29 +12,53 @@
...
         )
-
-        db.session.add(new_user)
+        db.session.add(user)
         db.session.commit()
+        return '<h1>User added</h1>'
+    return '<h1>API/User endpoint -- Try using Postman</h1>'
 
-        return 'OK', 200
+@user_blueprint.route('/api/book', methods=['GET','POST'])
```
Comment (OscarSierra24) [inline app/routes.py:32]: Puedes crear otro blueprint para book?

File: app/routes.py [hunk 2]
```diff
@@ -11,29 +12,53 @@
...
+            request.form["location"],
+            request.form["university"],
+            int(request.form["semester"]),
+            request.form["major"]
         )
-
-        db.session.add(new_user)
+        db.session.add(user)
         db.session.commit()
+        return '<h1>User added</h1>'
```
Comment (OscarSierra24) [inline app/routes.py:29]: hay que regresar un json con los status apropiados de http (en el caso exitoso 200)
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200
https://en.wikipedia.org/wiki/List_of_HTTP_status_codes

File: app/__init__.py [hunk 3]
```diff
@@ -1,10 +1,9 @@
-from flask import Flask
+from flask import Flask, url_for, request, redirect, send_file, Blueprint, flash, send_from_directory, after_this_request
```
Comment (OscarSierra24) [inline app/__init__.py:1]: son necesarios estos nuevos imports?

File: app/models/listing.py [hunk 4]
```diff
@@ -2,18 +2,19 @@
...
     no_available = db.Column(db.Integer)
     price = db.Column(db.Float)
-    user_id = db.Column(db.Integer, db.ForeignKey('book.id'), nullable=False)
-    book_id = db.Column(db.Integer, db.ForeignKey('user.id'), nullable=False)
+    user_id = db.Column(db.Integer, db.ForeignKey('user.id'), nullable=False)
+    book_id = db.Column(db.Integer, db.ForeignKey('book.id'), nullable=False)
 
-    def __init__(self, photo, description, condition, no_available, price,
-                 semester, major):
+    def __init__(self, photo, description, condition, no_available, price, user_id, book_id):
```
Comment (OscarSierra24) [inline app/models/listing.py:13]: no es buena práctica tener lineas mayores de 80 caracteres, deberías de hacer un salto de linea antes de los 80 caracteres e indentar como PEP8 lo indica: continuar escribiendo a partir de donde se abre el paréntesis

Comment (OscarSierra24) [inline app/models/listing.py:13]: no es buena práctica tener lineas con más de 80 caracteres, es mejor que nos apeguémos a los guidelines de PEP8, hacer un salto de linea e indentar a partir de donde se abrió el paréntesis

File: app/__init__.py (hunk 3 above)
Comment (JorgePS14) [reply app/__init__.py:1]: No, lo eran cuando los routes estaban en el init, deja lo cambio.

**Pull request merged**